ibis/pyspark/tests/test_window.py

```diff
@@ -0,0 +1,93 @@
+import pandas.util.testing as tm
+import pyspark.sql.functions as F
+import pytest
+from pyspark.sql.window import Window
+from pytest import param
+
+import ibis
+
+pytest.importorskip('pyspark')
+pytestmark = pytest.mark.pyspark
+
+
+@pytest.mark.parametrize(
+    ('ibis_window', 'spark_range'),
+    [
+        param(
+            ibis.trailing_window(
+                preceding=ibis.interval(hours=1),
+                order_by='time',
+                group_by='key',
+            ),
+            (-3600, 0),
+        ),
+        param(
+            ibis.trailing_window(
+                preceding=ibis.interval(hours=2),
+                order_by='time',
+                group_by='key',
+            ),
+            (-7200, 0),
+        ),
+        param(
+            ibis.range_window(
+                preceding=0,
+                following=ibis.interval(hours=1),
+                order_by='time',
+                group_by='key',
+            ),
+            (0, 3600),
+        ),
+    ],
+)
+def test_time_indexed_window(client, ibis_window, spark_range):
+    table = client.table('time_indexed_table')
+    result = table.mutate(
+        mean=table['value'].mean().over(ibis_window)
+    ).compile()
+    result_pd = result.toPandas()
+    spark_table = table.compile()
+    spark_window = (
+        Window.partitionBy('key')
+        .orderBy(F.col('time').cast('long'))
+        .rangeBetween(*spark_range)
+    )
+    expected = spark_table.withColumn(
+        'mean', F.mean(spark_table['value']).over(spark_window),
+    ).toPandas()
+    tm.assert_frame_equal(result_pd, expected)
+
+
+# TODO: multi windows don't update scope correctly
+@pytest.mark.xfail
```

Comment:
can you add this int he reason for the xfail and create an issue for this (and list it here)

Reply:
+1

Reply:
np, this is just the next issue I am going to fix, https://github.com/ibis-project/ibis/issues/2412